File: WWF/WF_WCF_Samples v4.0/WCF/Basic/Ajax/PostAjaxService/VB/service/service.vb
﻿'----------------------------------------------------------------
' Copyright (c) Microsoft Corporation.  All rights reserved. 
'----------------------------------------------------------------

Imports System
Imports System.ServiceModel

Namespace Microsoft.Samples.PostAjaxService
    ' Define a service contract.
    <ServiceContract([Namespace]:="PostAjaxService")> _
    Public Interface ICalculator
        <OperationContract()> _
        Function Add(ByVal n1 As Double, ByVal n2 As Double) As Double
        <OperationContract()> _
        Function Subtract(ByVal n1 As Double, ByVal n2 As Double) As Double
        <OperationContract()> _
        Function Multiply(ByVal n1 As Double, ByVal n2 As Double) As Double
        <OperationContract()> _
        Function Divide(ByVal n1 As Double, ByVal n2 As Double) As Double
    End Interface

    Public Class CalculatorService
        Implements ICalculator

        Public Function Add(ByVal n1 As Double, ByVal n2 As Double) As Double Implements ICalculator.Add
            Return n1 + n2
        End Function

        Public Function Subtract(ByVal n1 As Double, ByVal n2 As Double) As Double Implements ICalculator.Subtract
            Return n1 - n2
        End Function

        Public Function Multiply(ByVal n1 As Double, ByVal n2 As Double) As Double Implements ICalculator.Multiply
            Return n1 * n2
        End Function

        Public Function Divide(ByVal n1 As Double, ByVal n2 As Double) As Double Implements ICalculator.Divide
            Return n1 / n2
        End Function
    End Class

End Namespace

File: WCF/Windows SDK WCF Samples/TechnologySamples Beta 1/Basic/Binding/Net/Tcp/PortSharing/CS/service/service.cs

//  Copyright (c) Microsoft Corporation.  All Rights Reserved.

using System;
using System.ServiceModel;

namespace Microsoft.ServiceModel.Samples
{
   // Define a service contract
   [ServiceContract(Namespace = "http://Microsoft.ServiceModel.Samples")]
   public interface ICalculator
   {
      [OperationContract]
      double Add(double n1, double n2);
      
      [OperationContract]
      double Subtract(double n1, double n2);
      
      [OperationContract]
      double Multiply(double n1, double n2);
      
      [OperationContract]
      double Divide(double n1, double n2);
   }
   
   // Service class that implements the service contract
   public class CalculatorService : ICalculator
   {
      public double Add(double n1, double n2) { return n1 + n2; }
      public double Subtract(double n1, double n2) { return n1 - n2; }
      public double Multiply(double n1, double n2) { return n1 * n2; }
      public double Divide(double n1, double n2) { return n1 / n2; }
   }

   class service
   {
      static void Main(string[] args)
      {
         // Configure a binding with TCP port sharing enabled
         NetTcpBinding binding = new NetTcpBinding();
         binding.PortSharingEnabled = true;
         
         // Start a service on a fixed TCP port
         ServiceHost host = new ServiceHost(typeof(CalculatorService));
         ushort salt = (ushort)new Random().Next();
         string address = String.Format("net.tcp://localhost:5555/calculator/{0}", salt);
         host.AddServiceEndpoint(typeof(ICalculator), binding, address);
         host.Open();
         Console.WriteLine("Service #{0} listening on {1}.", salt, address);
         Console.WriteLine("Press <ENTER> to terminate service.");
         Console.ReadLine();
         host.Close();
      }
   }
}
